src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -94,6 +97,24 @@ public async Task<Compilation> CompileAsync(string inputPath, bool skipRestore)
             return compilation;
         }
 
+
+        public ProgramSyntax CompileParams(string inputPath, bool skipRestore)
+        {
+            var fileText = File.ReadAllText(inputPath);
+
+            var lineStarts = TextCoordinateConverter.GetLineStarts(fileText);
+            var inputUri = PathHelper.FilePathToFileUrl(inputPath);
+
+            var parser = new ParamsParser(fileText);
+            var syntax = parser.Program();
+
+            foreach(var diagnostic in syntax.GetParseDiagnostics()){
+                diagnosticLogger.LogDiagnostic(inputUri, diagnostic, lineStarts);
+            };
+
+            return syntax;
```

Comment:
Might be worth creating the parameters file equivalent of the `Compilation` class at this point - just to simplify the passing around of `syntax`.

Reply:
I plan to sort out creation of Compilation class with Cathy sometime today